bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/enable/HasLanguageServerPropertyTester.java

```diff
@@ -26,9 +38,49 @@ public HasLanguageServerPropertyTester() {
 	@Override
 	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
 		if (cLanguageServerProvider != null) {
-			return cLanguageServerProvider.isEnabledFor(receiver);
+			if (receiver instanceof URI) {
+				// called from the language server enabler for LSP4E:
+				var uri = (URI) receiver;
+				var fileHandle = getFileHandle(uri);
+				if(fileHandle != null && fileHandle instanceof IResource) {
+					var resource = (IResource) fileHandle;
+					if (resource != null && resource.getProject() != null) {					
+						return cLanguageServerProvider.isEnabledFor(resource.getProject());
+					}
+				}
+				// when resource == null it's an external file: Check if the file is already opened, if not check the active editor:
+				return LspUtils.isFileOpenedInLspEditor(uri);
+			} else if (receiver instanceof ITranslationUnit) {
+				// called to enable the LS based CSymbolsContentProvider: 
+				return Optional.of((ITranslationUnit) receiver)
+						 .map(ITranslationUnit::getCProject)
+						 .map(ICProject::getProject)
+						 .map(cLanguageServerProvider::isEnabledFor)
+						 .orElse(Boolean.FALSE);
+			} else if (receiver instanceof DocumentSymbolWithFile) {
+				// called to enable the LS based CSymbolsContentProvider:
+				return true;
+			} else if (receiver instanceof IProject) {
```

Comment:
What if we have an `IFile` here? Not sure if we can have 'IFolder'.
Actually this is where we should use `Adapters.adapt`

Reply:
Yes, you are right. But I don't think that this case is needed anymore. I'll remove it.